firebase-firestore/src/test/java/com/google/firebase/firestore/local/LocalStoreTestCase.java

```diff
@@ -963,9 +964,8 @@ public void testReadsAllDocumentsForInitialCollectionQueries() {
     localStore.executeQuery(query, /* usePreviousResults= */ true);
 
     assertRemoteDocumentsRead(/* byKey= */ 0, /* byQuery= */ 2);
-    if (!Persistence.OVERLAY_SUPPORT_ENABLED) {
```

Comment:
Not a blocker, but we should ideally remove all statements like this (including the flag) or leave them all here. If we leave the flag, we should be able to turn it off during development and have the tests pass.